src/Authenticator/PasetoAuthenticator.php

```diff
@@ -0,0 +1,200 @@
+ * @link https://github.com/paseto-standard/paseto-spec
+ */
+class PasetoAuthenticator extends TokenAuthenticator
+{
+    public const LOCAL = 'local';
+    public const PUBLIC = 'public';
+
+    /**
+     * @inheritDoc
+     */
+    protected $_defaultConfig = [
+        'header' => 'Authorization',
+        'queryParam' => 'token',
+        'tokenPrefix' => 'bearer',
+        'returnPayload' => true,
+        'version' => null,
+        'purpose' => null,
+        'secretKey' => null,
+        //'subjectKey' => IdentifierInterface::CREDENTIAL_JWT_SUBJECT,
+    ];
+
+    /**
+     * Payload data.
+     *
+     * @var object|null
+     */
+    protected $payload;
+
+    /**
+     * @inheritDoc
+     */
+    public function __construct(IdentifierInterface $identifier, array $config = [])
+    {
+        parent::__construct($identifier, $config);
+
+        if ($this->getConfig('version') === null) {
+            $this->setConfig('version', new Version4());
+        }
+        if (!$this->getConfig('version') instanceof ProtocolInterface) {
+            throw new RuntimeException(
+                'PASETO `version` config must be an instance of ' . ProtocolInterface::class
+            );
+        }
+
+        if (!in_array($this->getConfig('purpose'), [self::PUBLIC, self::LOCAL])) {
+            throw new RuntimeException('PASETO `purpose` config must one of: local or public');
+        }
+
+        if (empty($this->getConfig('secretKey'))) {
+            if (!class_exists(\Cake\Utility\Security::class)) {
+                throw new RuntimeException('PASETO `secretKey` config must be defined');
+            }
+            $this->setConfig('secretKey', \Cake\Utility\Security::getSalt());
+        }
+    }
+
+    /**
+     * Authenticates the identity based on a PASETO token contained in a request.
+     *
+     * @link https://paseto.io/
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request that contains login information.
+     * @return \Authentication\Authenticator\ResultInterface
+     * @throws \Exception
+     */
+    public function authenticate(ServerRequestInterface $request): ResultInterface
+    {
+        try {
+            $result = $this->getPayload($request);
+        } catch (Exception $e) {
+            return new Result(
+                null,
+                Result::FAILURE_CREDENTIALS_INVALID,
+                [
+                    'message' => $e->getMessage(),
+                    'exception' => $e,
+                ]
+            );
+        }
+
+        if (!$result instanceof JsonToken) {
+            return new Result(null, Result::FAILURE_CREDENTIALS_INVALID);
+        }
+
+        if (empty($result->getSubject())) {
+            return new Result(null, Result::FAILURE_CREDENTIALS_MISSING);
+        }
+
+        if ($this->getConfig('returnPayload')) {
+            $array = array_merge(
+                $result->getClaims(),
+                ['footer' => $result->getFooterArray()]
```

Comment:
What is the footer used for by application logic?

Reply:
A token is split into version, purpose, encrypted data, and unencrypted data with dots as the separator.

```
v4.local.KQVqiaVJbg_V0W9s-ekQMYyjA5sZCSMUFw6WxLJ_6806uE7ceNad9ABgGwgJ_HDN8zLDC_SF9hzFoESX8lt-v_o1LjsahdoOETFlp884ztE-vz5MZq6CfvaVhH4cjT8Wi6p43H95jegzM3BhbjrJpAiUOWISPxbR5Da90WSkCUVSQfMjCzSrTjEQRdTlbBNpjzEOen6f8Vaw-Wl2eqtmzgVDZ5LDcyR9fatO_p_IhH8cFwLXILoORpjFBgeCDXZqr9jaum6zEvAs3XSfWB86RLUM_QD-elM_2DkaCwqS_1btNYKpahRkLpr6ehM8dSKL.eyJmb290ZXJfZGF0YSI6ImlzIHVuZW5jcnlwdGVkIGJ1dCB0YW1wZXIgcHJvb2YifQ
```

So you can base64 decode this: `eyJmb290ZXJfZGF0YSI6ImlzIHVuZW5jcnlwdGVkIGJ1dCB0YW1wZXIgcHJvb2YifQ` to: 

```
{"footer_data":"is unencrypted but tamper proof"}
```

I assume to give API clients or any client that doesn't have the secret the ability to read non-sensitive data.